rlog/2024-12-08-merkle-tree.mdx

```diff
@@ -0,0 +1,193 @@
+Further, we will assume that nodes in a binary tree are either a parent node with two children or a leaf.
+As strange as it sounds, each child node has exactly one parental node.
+
+![Image 2](/img/vac101/vac101_binary_tree.png)
+
+A binary tree with $2^n$ leaves consist of $n+1$ layers.
+Additionally, such a tree has $2^{n+1}-1$ nodes.
+
+## Merkle trees
+
+A Merkle tree is a specialized tree in which each of its nodes contain the evaluation of a hash function.
+Merkle trees are usually taken to have a binary tree structure.
+As such, the presentation we provide in this section will be for binary trees.
+
+### Construction
+In this section we show how Merkle tree are constructed to compress a digest $D$.
+Suppose that the digest $D$ consists of $2^n$ entries;
+we assume that the digest $D$ has this many entries since a Merkle tree is a binary tree.
+Additionally, each digest can be padded to ensure that $D$ has the desired number of entries.
+
+Each leaf of the Merkle tree contains the hash of a digest entry.
+Each parent node contains the hash of the concatenation of their children nodes.
+Through this iterative construction, we reach the root of the tree.
+The value contained in the root node is called the root hash.
+The root hash is a compressed representation of the digest $D$.
+
+![Image 2](/img/vac101/vac101_merkle_tree.png)
+
+Each node in the Merkle tree is computed by taking a hash.
+Since a binary tree with $2^n$ leaves has $2^{n+1}-1$ nodes,
+then we need to evaluate $2^{n+1}-1$ hashes to construct the Merkle tree.
+
+### Merkle tree intregity
+
+A large quantity of data can be compressed to a single hash value.
+A natural question to ask is: could a clever party find another digest that yields a Merkle tree with the same root hash?
+If possible, this would compromise the ledger since the blockchain's history could be altered.
+Fortunately, Merkle trees are quite secure.
+In fact, Merkle trees can be used to both bind and hide a digest.
+
+The Merkle tree is able to bind a digest with one of the properties of hash functions (see our previous Vac-101 [[2](https://vac.dev/rlog/vac101-fiat-shamir#hash-functions)] for information on hash functions).
+A hash function is collision resistant; it is difficult for a malicious party to find two values share the same hash value.
+This collision resistance property, essentially, fixes the input to each leaf and into their parent, their parent's parent, and so on.
+
+In certain applications,
+it may be desirable for the digest of a Merkle tree to be kept confidential.
+This is achieved with the preimage resistant propery of hash functions.
+A hash function is preimage resistant provided that it is difficult reverse the hashing operation. 
+It would be necessary for a malicious party to find preimages to each node starting from the root node to determine original digest.
+
+Now, we see that Merkle trees are secured structures that are tamper resistant.
+
+### Proof of membership
```

Comment:
I would clarify that the verification process needs the root hash to be known from a trusted source, if I'm not wrong.